cmake/FindLibUnwind.cmake

```diff
@@ -0,0 +1,86 @@
+#[========================================================================[.rst:
+FindLibUnwind
+--------
+Finds the libunwind library.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+``LIBUNWIND_FOUND``
+  True if the system has the libunwind library.
+``LIBUNWIND_VERSION``
+  The version of the libunwind library which was found.
+``LIBUNWIND_INCLUDE_DIR``
+  Include directory needed to use libunwind.
+``LIBUNWIND_LIBRARIES``
+  Libraries needed to link to libunwind.
+
+Cache Variables
+^^^^^^^^^^^^^^^
+``LIBUNWIND_INCLUDE_DIR``
+  The directory containing ``libunwind.h``.
+``LIBUNWIND_LIBRARIES``
+  The paths to the libunwind libraries.
+#]========================================================================]
+
+include(FindPackageHandleStandardArgs)
+include(GetLibUnwindVersion)
+
+find_package(PkgConfig QUIET)
+pkg_check_modules(PC_LIBUNWIND QUIET libunwind)
+
+find_path(LIBUNWIND_INCLUDE_DIR libunwind.h ${PC_LIBUNWIND_INCLUDE_DIRS})
+if(LIBUNWIND_INCLUDE_DIR)
+    include_directories(${LIBUNWIND_INCLUDE_DIR})
+endif()
+
+if(BUILD_STATIC AND NOT APPLE)
+    set(LIBUNWIND_LIBRARY_NAME libunwind.a)
+else()
+    # Only a dynamic version of libunwind is available on macOS: also, we
+    # should link against the umbrella framework `System` — otherwise `ld` will
+    # complain that it cannot link directly with libunwind.tbd.
+    set(LIBUNWIND_LIBRARY_NAME System unwind)
+endif()
+find_library(LIBUNWIND_LIBRARY NAMES ${LIBUNWIND_LIBRARY_NAME}
+             PATHS ${PC_LIBUNWIND_LIBRARY_DIRS})
+
+if(APPLE)
+    set(LIBUNWIND_LIBRARIES ${LIBUNWIND_LIBRARY})
+else()
+    if(BUILD_STATIC)
+        set(LIBUNWIND_PLATFORM_LIBRARY_NAME
+            "libunwind-${CMAKE_SYSTEM_PROCESSOR}.a")
```

Comment:
[Ditto](https://github.com/tarantool/tarantool/pull/6889#discussion_r829986766).

Prior 7dc9fe44 to this was processed as follows:
https://github.com/tarantool/tarantool/blob/dbaf03dc953905a74c92ab2e3792a71c30b0ed0e/cmake/compiler.cmake#L167-L171

The solution you suggest will not work, please see documentation for find_library: https://cmake.org/cmake/help/latest/command/find_library.html

> Each library name given to the `NAMES` option is first considered as a library file name and then considered with platform-specific prefixes (e.g. `lib`) and suffixes (e.g. `.so`). Therefore one may specify library file names such as `libfoo.a` directly. This can be used to locate static libraries on UNIX-like systems.

Please see the order in which CMake processes library suffixes (e.g., the first library it will find): https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_LIBRARY_SUFFIXES.html
```cmake
message(STATUS "CMAKE_FIND_LIBRARY_SUFFIXES=${CMAKE_FIND_LIBRARY_SUFFIXES}")
```
Output:
```console
CMAKE_FIND_LIBRARY_SUFFIXES=.so;.a
```

This implies that by default a dynamic version of the library will be found — one needs to explicitly specify the full library name in order to find the static version.

Last, but not least, please see other examples in our repository:
https://github.com/tarantool/tarantool/blob/7dc9fe44a5aebc2b44c240221d2a71cd6d97b6b0/cmake/FindCURL.cmake#L19-L25

https://github.com/tarantool/tarantool/blob/7dc9fe44a5aebc2b44c240221d2a71cd6d97b6b0/cmake/FindICU.cmake#L25-L33

Etc. — all FindLibrary.cmake scripts do it exactly the same way.

Reply:
ok, looks reasonable